[bazel][jdk7][jre7][jdk7][jre8][p4broker][libsasl2] - Deprecating plans

MindNumbing opened the pull request:
This commit moves deprecated plans to the deprecated folder as detailed [here](https://github.com/habitat-sh/core-plans/blob/master/docs/dev/policy_documents/deprecating-packages.md).

Signed-off-by: Steven Marshall <SMarshall@chef.io>

MindNumbing commented:
These additional pre-requisites for deprecation should be added to the policy documents regarding deprecating packages.

stevendanna commented:
> These additional pre-requisites for deprecation should be added to the policy documents regarding deprecating packages.

I see the first request as just trying to maintain our desired practice of commit messages for important changes including the "why" behind the change.  I'm not sure we need to add it to the policy for deprecations specifically because it applies to all non-trivial changes we make to code I think.

I agree that the point about providing user guidance and making sure that we clean up or document any packages that end up depending on the deprecated packages in core _should_ be added to the RFC for deprecating plans.  Do you think these would be reasonable additions or do you think they are unnecessary?  Feel free to push back if you think I've made an unreasonable request.

MindNumbing commented:
No, I was not disagreeing with you. They were good suggestions.

I have:
Added reasoning to the original commit message.
Marked each README with notice and explaination of deprecation
Suggested alternatives where possible (to the equivalent corretto build) 
Changed gocd-server to use corretto (this had already been done on the refresh branch).

Similarly, once the refresh is completed there will be no plans making use of JRE/JDK instead using corretto builds.

predominant commented:
Why are we deprecating rethinkdb? I use that on a number of projects.

MindNumbing commented:
Rethinkdb seems to be pinned to core/gcc/5.2.0 which is a very old version of core/gcc. That and the fact that bldr wasn't returning any rdeps for RethinkDB being used, I assumed that it wasn't being activly used.

stevendanna commented:
I've been thinking about this a bit and am wondering if we should invest in getting `bazel` working again because it is a common build tool.

stevendanna commented:
I'm closing this as it appears to have gone in with the refresh.

I'll open a new PR to un-deprecate bazel sometime in the next few weeks.
